cpp/src/barretenberg/stdlib/primitives/field/field.cpp

```diff
@@ -537,32 +551,38 @@ template <typename ComposerContext> field_t<ComposerContext> field_t<ComposerCon
 
 template <typename ComposerContext> void field_t<ComposerContext>::assert_is_zero(std::string const& msg) const
 {
-    if (get_value() != barretenberg::fr(0)) {
```

Comment:
For circuit simulation, we want to be able to create failing circuits. The `ASSERT` below prevents doing that, so I needed an `if` `else`.

Reply:
We see this happen in various other places.